Added a Home Screen and the ability to add tasks

Rift3000 opened the pull request.

hanskokx commented on lib/models/task.dart:
```diff
@@ -8,3 +10,10 @@ class Task {
+  final String title;
+  Icon taskIcon = Icon(Icons.alarm);
+
+  TaskList({this.title, this.taskIcon});
```
If `title` is empty, this will fail.  Mark `title` as required.

Rift3000 replied:
Good point

hanskokx commented on lib/models/task_home_data.dart:
```diff
@@ -0,0 +1,25 @@
+class TaskListHome extends ChangeNotifier {
+  List<TaskList> taskList = [
+    TaskList(title: "Rift's Corner", taskIcon: Icon(Icons.alarm)),
+    TaskList(title: "Han's Corner", taskIcon: Icon(Icons.run_circle)),
```
Han's => Hans'

Rift3000 replied:
ahh I see

hanskokx commented on lib/models/task_home_data.dart:
```diff
@@ -0,0 +1,25 @@
+  }
+
+  void addTask(String myTaskTitle) {
+    final task = TaskList(title: myTaskTitle);
```
Will you add the `taskIcon` later?

Rift3000 replied:
I thinnk that is something that could be explored in the future. Allowing the user to select a task icon would be cool

hanskokx commented on lib/screens/add_task_screen.dart:
```diff
@@ -45,9 +46,11 @@ class AddTaskScreen extends StatelessWidget {
+                    .addTask(newTaskTitle);
                 Navigator.pop(context);
               },
+              style: TextButton.styleFrom(backgroundColor: Colors.lightBlue),
```
Nit: Perhaps the color should be specified in a global theme?  Then you can use `Theme.of(context).xxxx`

Rift3000 replied:
mmhhnnn you smart

hanskokx commented on lib/screens/home.dart:
```diff
@@ -0,0 +1,143 @@
+                              style: TextStyle(
+                                  color: Colors.white,
+                                  fontSize:
+                                      MediaQuery.of(context).size.width / 20,
```
This seems like a dangerous calculation.  Maybe set a minimum, and use a `Wrap()`?

Rift3000 replied:
Set it to a minimum? what do you mean?

hanskokx replied:
Right now, the screen size always determines the font size. If someone happens to have a very small screen, the text could be size 2.  Just do a simple if statement. If the resulting font size is smaller than ~10, use a font size of 10. Otherwise, make sure you are using a whole number.

hanskokx commented on lib/screens/home.dart:
```diff
@@ -0,0 +1,143 @@
+      child: Text('Add A Task',
+          style: TextStyle(
+              color: Colors.white,
+              fontSize: MediaQuery.of(context).size.width / 20,
```
Same here

hanskokx commented on lib/screens/home.dart:
```diff
@@ -0,0 +1,143 @@
+          cursorColor: Colors.white,
+          autocorrect: true,
+          onChanged: (newTitle) {
+            print(newTitle);
```
Don't forget to remove debugging statements

Rift3000 replied:
totally forgot

hanskokx commented on lib/screens/home.dart:
```diff
@@ -0,0 +1,143 @@
+              elevation: 5.0,
+              padding: EdgeInsets.all(20.0)),
+          onPressed: () {
+            Provider.of<TaskListHome>(context, listen: false).addTask(title);
```
If this fails for some reason, we'll still pop.  Might want to introduce some additional error handling, especially if this ends up pushing to a network resource (which may be unavailable)

hanskokx reviewed (changes requested):
I didn't try compiling and running this - I'm making the assumption that you wouldn't push it if it didn't compile.

Rift3000 commented:
I addressed all the issues you highlighted and added two images to the Readme to show a little progress. Issues touched: Error handling, using ThemeData, debugging statements removed, etc. Currently don't see the need to add a wrap. We could possibly add it or review in the future.

hanskokx commented on lib/screens/home.dart:
```diff
@@ -97,47 +98,100 @@ class TaskHome extends StatelessWidget {
-      ],
-    ),
-  );
+      backgroundColor: Colors.purple,
```
Maybe pull from the theme?

hanskokx commented on lib/screens/home.dart:
```diff
@@ -97,47 +98,100 @@ class TaskHome extends StatelessWidget {
+                    title = newTitle;
+                  },
+                  textAlign: TextAlign.center,
+                  style: TextStyle(color: Colors.black, fontSize: 20.0),
```
Theme?

hanskokx commented on lib/screens/home.dart:
```diff
@@ -97,47 +98,100 @@ class TaskHome extends StatelessWidget {
+                  style: TextStyle(color: Colors.black, fontSize: 20.0),
+                  decoration: InputDecoration(
+                      filled: true,
+                      fillColor: Colors.white,
```
Theme?....you get the idea....

Rift3000 replied:
lol Really not seeing the point of changing every color to use theme. Enlighten me

hanskokx replied:
Light mode vs Dark mode

Rift3000 replied:
Makes sense. Adding a dark mode later would be cool

hanskokx replied:
Also, it keeps things consistent. If all the text should be the same color, using a shared variable saves time and effort when you want to change it.

Rift3000 replied:
Ok, I used Theme

hanskokx reviewed (approved):
Tiny changes, but otherwise looks good.

hanskokx commented on lib/screens/home.dart:
```diff
@@ -0,0 +1,198 @@
+          "Create",
+          style: TextStyle(color: Theme.of(context).canvasColor),
+        ),
+        backgroundColor: Colors.lightBlueAccent,
```
Theme 😅

Rift3000 replied:
HOW did I miss these!!! LOL

hanskokx replied:
n00b

Rift3000 replied:
lol..... takes one to know one.....

hanskokx commented on lib/screens/home.dart:
```diff
@@ -0,0 +1,198 @@
+              clipBehavior: Clip.none,
+              children: <Widget>[
+                TextFormField(
+                  cursorColor: Colors.black,
```
🤔

hanskokx commented on lib/screens/home.dart:
```diff
@@ -0,0 +1,198 @@
+                          ? IconButton(
+                              icon: Icon(
+                                Icons.error,
+                                color: Colors.red,
```
... I'll accept it

hanskokx commented on lib/screens/home.dart:
```diff
@@ -0,0 +1,198 @@
+                          width: 270,
+                          height: 50,
+                          decoration: BoxDecoration(
+                              color: Colors.white,
```
🤔

hanskokx commented on lib/screens/home.dart:
```diff
@@ -0,0 +1,198 @@
+            ),
+            ElevatedButton(
+              style: ElevatedButton.styleFrom(
+                  primary: Colors.lightBlue,
```
🤔

hanskokx commented on lib/screens/home.dart:
```diff
@@ -0,0 +1,198 @@
+                }
+              },
+              child: Text('Add Task',
+                  style: TextStyle(color: Colors.white, fontSize: 20.0)),
```
🤔

hanskokx reviewed (changes requested):
So close! You can do it, Jason! 📣

Rift3000 commented:
This time for sure! None escaped my sight

Rift3000 commented:
Why not merge the pull request

hanskokx commented:
> Why not merge the pull request

Because I don't have permissions on this repo 🤷🏻‍♂️
